src/input.rs

```diff
@@ -157,25 +157,55 @@ impl<'a> Input<'a> {
         &mut self.description
     }
 
-    pub(crate) fn open<R: BufRead + 'a>(self, stdin: R) -> Result<OpenedInput<'a>> {
+    pub(crate) fn open<R: BufRead + 'a>(
+        self,
+        stdin: R,
+        stdout_identifier: Option<&Identifier>,
+    ) -> Result<OpenedInput<'a>> {
         let description = self.description().clone();
         match self.kind {
-            InputKind::StdIn => Ok(OpenedInput {
-                kind: OpenedInputKind::StdIn,
-                description,
-                metadata: self.metadata,
-                reader: InputReader::new(stdin),
-            }),
+            InputKind::StdIn => {
+                if let Some(stdout) = stdout_identifier {
+                    let input_identifier = Identifier::try_from(clircle::Stdio::Stdin)
+                        .map_err(|e| format!("Stdin: Error identifying file: {}", e))?;
+                    if stdout.surely_conflicts_with(&input_identifier) {
+                        return Err(format!("Stdin: input file is output file",).into());
```

Comment:
I know it was similar before, but maybe we could extend this error message a little bit. After all, that's everything that a user is ever going to see from this feature. When is this error triggered? If someone calls `bat < file > file`?

Reply:
Good point, gotta beat coreutils in some way :sweat_smile:

Reply:
:smile: